### ISSUE TITLE
Allow Go Comments outside of Block definitions in layouts [BUG]

### ISSUE THREAD

**Issue by HenrySkup** [gohugoio/hugo#6820]
Hi, This is in regards to [this thread on Discourse](https://discourse.gohugo.io/t/hugo-0-63-1-does-not-render-whole-site/23000).

Basically, v0.63.1 (and not v0.63.2) are only rendering a fraction of the pages that it should be.

@bep and @onedrawingperday I've shared my repo with you, please let me know if I can help out .

----

### What version of Hugo are you using?
> v0.63.1
### Does this issue reproduce with the latest release?
yes

**Comment by HenrySkup** [comment]
Just checked with another site, same thing is happening.  Here is the output:

### v0.62.2
```
> hugo server -D --disableFastRender --gc

                   | EN
+------------------+-----+
  Pages            |  55
  Paginator pages  |   0
  Non-page files   |   0
  Static files     |   0
  Processed images |   2
  Aliases          |   0
  Sitemaps         |   1
  Cleaned          | 619

Built in 521 ms
```

### v0.63.2

```
> hugo server -D --disableFastRender --gc

                   | EN
-------------------+-----
  Pages            | 50
  Paginator pages  |  0
  Non-page files   |  0
  Static files     |  0
  Processed images |  0
  Aliases          |  0
  Sitemaps         |  1
  Cleaned          |  0

Built in 850 ms
```

**Comment by HenrySkup** [comment]
Ok, so I think i figured this out…

Tried fiddling around with the new(?) template lookup order but that didn’t seem to be the issue.

Then I just deleted some Go Template comments that were at the top of my layouts/***.html files and, bam, things are working now. These have been there for a while, some i’m not sure what had changed. I know there is an issue with HTML comments, but go comments should (ideally) be fine?

I'm renaming this issue.
